[initramfs] Add initramfs hook to log more info for blockdev wait issue in booting

byu343 opened the pull request:
<!--
     Please make sure you've read and understood our contributing guidelines:
     https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

     ** Make sure all your commits include a signature generated with `git commit -s` **

     If this is a bug fix, make sure your description includes "fixes #xxxx", or
     "closes #xxxx" or "resolves #xxxx"

     Please provide the following information:
-->

#### Why I did it
This change helps to the debug the issue when the block device for /host sometimes fails to initialize in booting.
When the issue happened before, the issue was not persistent so hard to reproduce, and the watchdog was triggered so there was no enough log to understand the cause.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Add an Arista-specific initramfs hook to wait for block dev for /host
 - Skip it for non-Arista hardware
 - Skip it if UUID is not in /proc/cmdline
 - Skip if in kdump enviroment
 - Wait for the device based on UUID and the timeout is around 10 seconds
 - If failed, print the last 100 lines of dmesg, ls /dev and blkid to the console. 

#### How to verify it
Verified that the expected debug info can be printed on a device with modified wrong UUID
Verified that the normal device can boot without issue.

<!--
If PR needs to be backported, then the PR must be tested against the base branch and the earliest backport release branch and provide tested image version on these two branches. For example, if the PR is requested for master, 202211 and 202012, then the requester needs to provide test results on master and 202012.
-->

#### Which release branch to backport (provide reason below if selected)

<!--
- Note we only backport fixes to a release branch, *not* features!
- Please also provide a reason for the backporting below.
- e.g.
- [x] 202006
-->

- [ ] 202205
- [ ] 202211
- [ ] 202305
- [ ] 202311
- [ ] 202405
- [ ] 202411
- [x] 202505

#### Tested branch (Please provide the tested image version)

<!--
- Please provide tested image version
- e.g.
- [x] 20201231.100
-->

- [ ] <!-- image version 1 -->
- [ ] <!-- image version 2 -->

#### Description for the changelog
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->

<!--
 Ensure to add label/tag for the feature raised. example - PR#2174 under sonic-utilities repo. where, Generic Config and Update feature has been labelled as GCU.
-->

#### Link to config_db schema for YANG module changes
<!--
Provide a link to config_db schema for the table for which YANG model
is defined
Link should point to correct section on https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-yang-models/doc/Configuration.md
-->

#### A picture of a cute animal (not mandatory but encouraged)

byu343 committed:
Add initramfs hook to debug blockdev wait issue in booting

Signed-off-by: Boyang Yu <byu@arista.com>

Ryangwaite commented on files/initramfs-tools/arista-wait-blockdev:
```diff
@@ -0,0 +1,50 @@
+[ -z "$root_val" ] && exit 0
+
+wait_blockdev_retry_count=1
+while [ "$wait_blockdev_retry_count" -le 10 ]; do
```
In the worst case scenario, it looks like this will delay LACP/BGP restoration by 10s in the case of warmboot.

Can we avoid this change for warmboot? Or alternatively cap its runtime to ~1s worst case?

Good change for coldboot though.

byu343 replied:
This change repeatedly probes for /host every second. If the device does not appear, the warm-boot process would already be unable to proceed even without this modification.  In essence, the change only adds debug logging when the wait for /host takes too long, and it should not negatively impact cases where the device is slow to appear.

It helps to print some debug info we want to collect but could not because of the watchdog to power-cycle would kick in when block device not showing up for 180 seconds.
 - ls /dev
 - dmesg | tail -n 100
 - blkid

I also made the change less intrusive by:
- Removing set -e so the script continues even if a command like dmesg fails
- Ensuring the script always exits with status 0
- Changing the maximum wait time to 30 seconds. This means: If the device for /host take X (less than 30) seconds to appear, the script would block booting by X seconds for waiting for it. If it takes longer than 30 seconds, the script prints debug information, and then there is some existing system-level initramfs mechanism continues waiting for the device until it shows up.

mssonicbld commented:
This PR has backport request for branch(es): 202505.<br>Added label(s) for branch(es) 202505.

<div align="right"><sub>---Powered by SONiC BuildBot</sub></div>

byu343 committed:
Remove set -e and change max wait to 30 seconds

This change is only to print debug info, so it should not return
error and always return 0
Change max wait to 30 seconds for less chance of affecting warm-reboot.

Signed-off-by: Boyang Yu <byu@arista.com>

StormLiangMS reviewed (approved):
LGTM — net-positive: silently masks the slow-probe case on healthy boots, captures real evidence (`dmesg`/`ls /dev`/`blkid`) before the watchdog fires on the failure path. `exit 0` on every path means it cannot make boot worse than today.

One question on backport scope before we widen the blast radius:

**Do we actually need this on 202511 and 202505?** The 202605 backport makes sense — that's where ongoing diagnosis happens. But for the older release branches:

1. Have we **seen this bug occur on 202505 / 202511 in fleet**, or is the backport just precautionary? If it's only ever reproduced on master/202605, dropping the older backports keeps the change surface tight.
2. If we do backport, please also mirror `arista-hook`'s existing skip for warm/fast reboot so we don't add up to 30s to warmboot LACP/BGP recovery on a slow-probe box:
   ```sh
   SONIC_BOOT_TYPE=warm*|SONIC_BOOT_TYPE=fast*)
       exit 0
       ;;
   ```
   The earlier thread argued "warmboot is already toast if blockdev is missing" — true only if the device **never** appears; if it appears at second 15 of the wait, this script adds 15s to warmboot wall-clock that wasn't there before. Worth gating, especially for the 202505/202511 branches that are actively running warmboot in production.
